lib/intl.ts

```diff
@@ -90,7 +90,9 @@ export function DateTimeFormat(
   if (hasOptions) {
     const clonedResolved = ObjectAssign({}, ro);
     for (const prop in clonedResolved) {
-      if (!ES.HasOwnProperty(options, prop)) delete clonedResolved[prop];
+      if (!Object.prototype.hasOwnProperty.call(options, prop)) {
```

Comment:
Should save `Object.prototype.hasOwnProperty` at the top of the file in order to be robust against user code deleting it, and use a saved `Reflect.apply` to call it.